File: Melo/Melo.Models/Models/Artist/ArtistResponse.cs
namespace Melo.Models
{
	public class ArtistResponse
	{
		public int Id { get; set; }

		public string? Name { get; set; }

		public long? LikeCount { get; set; }

		public long? ViewCount { get; set; }

		public string? ImageUrl { get; set; }

		public List<GenreResponse> Genres { get; set; } = new List<GenreResponse>();
	}
}

File: Melo/Melo.UI/melo_desktop/lib/models/artist_response.dart
import 'package:melo_desktop/models/genre_response.dart';

class ArtistResponse {
  final int id;
  final String? name;
  final int? likeCount;
  final int? viewCount;
  final String? imageUrl;
  final List<GenreResponse> genres;

  ArtistResponse({
    required this.id,
    this.name,
    this.likeCount,
    this.viewCount,
    this.imageUrl,
    this.genres = const [],
  });

  factory ArtistResponse.fromJson(Map<String, dynamic> json) {
    return ArtistResponse(
      id: json['id'] as int,
      name: json['name'] as String?,
      likeCount: json['likeCount'] as int?,
      viewCount: json['viewCount'] as int?,
      imageUrl: json['imageUrl'] as String?,
      genres: (json['genres'] as List<dynamic>?)
              ?.map((e) => GenreResponse.fromJson(e as Map<String, dynamic>))
              .toList() ??
          [],
    );
  }
}
